src/main/kotlin/no/nav/familie/ks/sak/config/KafkaConfig.kt

```diff
@@ -108,6 +116,26 @@ class KafkaConfig(
         return consumerConfigs.toMap() + securityConfig()
     }
 
+    private fun consumerConfigsLatestAvro(): Map<String, Any> {
+        val kafkaBrokers = System.getenv("KAFKA_BROKERS") ?: "http://localhost:9092"
+        val schemaRegistry = System.getenv("KAFKA_SCHEMA_REGISTRY") ?: "http://localhost:9093"
+        val schemaRegistryUser = System.getenv("KAFKA_SCHEMA_REGISTRY_USER") ?: "mangler i pod"
+        val schemaRegistryPassword = System.getenv("KAFKA_SCHEMA_REGISTRY_PASSWORD") ?: "mangler i pod"
+        val consumerConfigs =
+            mutableMapOf(
+                ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG to kafkaBrokers,
+                "schema.registry.url" to schemaRegistry,
+                "basic.auth.credentials.source" to "USER_INFO",
+                "basic.auth.user.info" to "$schemaRegistryUser:$schemaRegistryPassword",
+                "specific.avro.reader" to true,
+                ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG to StringDeserializer::class.java,
+                ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG to KafkaAvroDeserializer::class.java,
+                ConsumerConfig.CLIENT_ID_CONFIG to "consumer-familie-ks-sak-2",
+                ConsumerConfig.AUTO_OFFSET_RESET_CONFIG to "latest",
```

Comment:
Det ble gjort et eksplisitt avgjørelse engang om:
ConsumerConfig.MAX_POLL_RECORDS_CONFIG to "1",
ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG to "false",

Om man bare hadde kommentert grunnen i koden, så hadde man sikkert husket hvorfor, men jeg tror det var for å tvinge Kafka til å gjøre commit mellom hver melding.  Med autocommit true, som i alle fall var default, så vil den ikke gjøre commit før poll(eller en bestemt tid). Om det lenger har noen betydning etter at man har gått over til en mer stabil AIVEN, og latest config er jeg usikker på. Tipper det går fint.

Reply:
Ahh fett, good to know!